### PR TITLE
Client Settings Revamp with Categories.

### PR DESCRIPTION
**Pull request by Arufonsu**
### About this Pull Request.
* Renamed "Options" to "Settings" in order to keep consistency.
* Added Tabs in order to categorize the client settings.
* "Tabs" are technically Buttons, but we will refer to them as "Tabs" for now, considering what they actually do, visually speaking.
* Game Settings has it's own tab. 
   (it is empty **for now**) 
   -  **Another PR will cover upcoming Game Settings, but first, we need to expand 
   the settings capabilities, see** #936 
   
* Video Settings has it's own tab.
* Audio Settings has it's own tab.
* Keybinding Settings (AKA: "Controls") has it's own tab.
* Apply button behaves as it usually does, but now saves the Keybinding Settings as well.
* When accessing the Settings Window from the Main or Escape Menus, Apply will save and take you back there.
* When accessing the Settings Window from the Main or Escape Menus, Cancel will take you back there without saving.
* When accessing the Settings Window by pressing a configured key, Apply will save and take you back to game.
* When accessing the Settings Window by pressing a configured key, Cancel will take you back to game without saving.

### Preview.
imagen

imagen

imagen

### About the required resources for this PR
* "settings_window.png" replaces "options_horizontal.png".
* "SettingsWindow.json" replaces "OptionsWindow.json".
* "options_dropdown.png" renamed to "settings_dropdown.png"
* Changes to EscapeMenu.json were made due to the "options" renaming.
* Changes to MenuWindow.json were made due to the "options" renaming.

[(Updated 16-10-2021) Download 'resources.zip' with the required files for this PR](https://github.com/AscensionGameDev/Intersect-Engine/files/7358016/resources.zip)
Added the sprites for tabs just in case now, they should be present in the latest asset pack since the addition of the chatbox tabs.
imagen

NOTE: This PR originally included the 3 new settings for entities info visibility, decided to exclude those for now, because they may still require some discussion and attention in a different PR afterwards, so expect more resource updates on this one !

### CODE REVIEW

**Comment by Arufonsu** [comment]
- Updated the pull request description.
- Updated the pull request preview and resources.
- Reverted the "back" button to "cancel", as previously discussed in discord with @Cheshire92

File: Intersect.Client/Interface/Shared/SettingsWindow.cs [hunk 1]
```diff
@@ -0,0 +1,726 @@
...
+            mSoundLabel.Text = Strings.Settings.SoundVolume.ToString((int) mSoundSlider.Value);
+
+            // SettingsWindow is not hidden anymore.
+            mSettingsPanel.IsHidden = false;
+
+            // Load every GUI element to their default state when showing up the settings window (pressed tabs, containers, etc.)
+            LoadSettingsWindow();
+        }
+
+        public bool IsVisible()
```
Comment (Cheshire92) [inline Intersect.Client/Interface/Shared/SettingsWindow.cs:570]: public bool IsVisible => !mSettingsPanel.IsHidden;

**Review by Cheshire92 — CHANGES_REQUESTED** [review]
I'm also not getting the tab graphics set up correctly when I download your resource zip?

Image

**Comment by Cheshire92** [comment]
Also just noticed that Cancel does not return to the escape menu in-game.

**Comment by Arufonsu** [comment]
> 
> 
> I'm also not getting the tab graphics set up correctly when I download your resource zip?
> 
> Image

do you have the latest resources that include the tabs for the chatboxes?
This are the ones being used for tabs, they don't really require any changes.
imagen

**Comment by Cheshire92** [comment]
Ah. I meant that there shouldn't be a difference in labels for the button, it should always be called cancel.. Not randomly cancel, back or exit. :P

However, it should still contextually behave properly.

File: Intersect.Client/Interface/Shared/SettingsWindow.cs (hunk 1 above)
Comment (Arufonsu) [reply Intersect.Client/Interface/Shared/SettingsWindow.cs]: [Done 👍🏽 ](https://github.com/AscensionGameDev/Intersect-Engine/pull/950/commits/a5eb8bb077c7beb9c4e46767ebb536997603a69f#diff-5c617a22ae8f6a814cb9e86128e65384b834c70141a564ded35be45f09da8512R586)

**Comment by Cheshire92** [comment]
It's honestly really annoying that you keep force pushing stuff as you're forcing me to re-review the entire thing and not just the changes since I can't tell what has or hasn't been changed like this. Could you please stop doing that?

**Comment by Cheshire92** [comment]
You've also broken my local version of this now and making me jump through all sorts of hoops to fix it. Well done.
Please never do that again. The easiest solution for me right now is to just delete my local repo and start over with that as a reset just tries to merge your old changes with your new ones somehow.

**Review by Cheshire92 — APPROVED** [review]

**Comment by Arufonsu** [comment]
> It's honestly really annoying that you keep force pushing stuff as you're forcing me to re-review the entire thing and not just the changes since I can't tell what has or hasn't been changed like this. Could you please stop doing that?


Oh.. my bad, didn't knew that you were working with a local repo, i just wanted the PR to be more in shape and less messy with small commits, as i was fixing small undesired things around.

I didn't really pictured how annoying that  could of been to be honest :( sorry, won't happen again.

Anyway.. thanks a lot for pushing and helping along with this PR, i hope that the idea and changes end up being useful for everyone when it comes to add new client settings :D !

 Regards ~

**Pull request merged**